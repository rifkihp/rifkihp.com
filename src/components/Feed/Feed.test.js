import React from 'react';
import renderer from 'react-test-renderer';

import Feed from './Feed';

const edge = s => ({
  node: {
    fields: {
      slug: s,
      categorySlug: s,
      dateFormatted: 'March 14, 2019',
    },
    frontmatter: {
      date: new Date('2019-03-14T12:00:00.000Z'),
      description: s,
      category: s,
      title: s,
    },
  },
});

describe('Feed', () => {
  const props = {
    edges: [edge('test_0'), edge('test_1')],
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Feed {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
