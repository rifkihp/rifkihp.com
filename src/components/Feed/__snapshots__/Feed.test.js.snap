// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div
  className="feed"
>
  <div
    className="feed__item"
  >
    <h2
      className="feed__item-title"
    >
      <Link
        className="feed__item-title-link"
        onClick={[Function]}
        to="test_0"
      >
        test_0
      </Link>
    </h2>
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime={2019-03-14T12:00:00.000Z}
      >
        March 14, 2019
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-category"
      >
        <Link
          className="feed__item-meta-category-link"
          to="test_0"
        >
          test_0
        </Link>
      </span>
    </div>
    <p
      className="feed__item-description"
    >
      test_0
    </p>
    <Link
      className="feed__item-readmore"
      onClick={[Function]}
      to="test_0"
    >
      Read
    </Link>
  </div>
  <div
    className="feed__item"
  >
    <h2
      className="feed__item-title"
    >
      <Link
        className="feed__item-title-link"
        onClick={[Function]}
        to="test_1"
      >
        test_1
      </Link>
    </h2>
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime={2019-03-14T12:00:00.000Z}
      >
        March 14, 2019
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-category"
      >
        <Link
          className="feed__item-meta-category-link"
          to="test_1"
        >
          test_1
        </Link>
      </span>
    </div>
    <p
      className="feed__item-description"
    >
      test_1
    </p>
    <Link
      className="feed__item-readmore"
      onClick={[Function]}
      to="test_1"
    >
      Read
    </Link>
  </div>
</div>
`;
